src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1150,38 +1121,8 @@ abstract class TContainer extends Type {
     )
   }
 
-  override def unsafeOrdering(missingGreatest: Boolean): UnsafeOrdering = {
-    val eltOrd = elementType.unsafeOrdering(missingGreatest)
-
-    new UnsafeOrdering {
-      override def compare(r1: MemoryBuffer, o1: Long, r2: MemoryBuffer, o2: Long): Int = {
-        val length1 = loadLength(r1, o1)
-        val length2 = loadLength(r2, o2)
-
-        var i = 0
-        while (i < math.min(length1, length2)) {
-          val leftDefined = isElementDefined(r1, o1, i)
-          val rightDefined = isElementDefined(r2, o2, i)
-
-          if (leftDefined && rightDefined) {
-            val eOff1 = loadElement(r1, o1, length1, i)
-            val eOff2 = loadElement(r2, o2, length2, i)
-            val c = eltOrd.compare(r1, eOff1, r2, eOff2)
-            if (c != 0)
-              return c
-          } else if (leftDefined != rightDefined) {
-            val c = if (leftDefined) -1 else 1
-            if (missingGreatest)
-              return c
-            else
-              return -c
-          }
-          i += 1
-        }
-        Integer.compare(length1, length2)
-      }
-    }
-  }
+  override def unsafeOrdering(missingGreatest: Boolean): CodifiedUnsafeOrdering =
+    new ContainerUnsafeOrdering(this, elementType.unsafeOrdering(missingGreatest), missingGreatest)
```

Comment:
I agree the missingGreatest seems like it's only relevant to compound data, and leads to strange code like this, perhaps best we discuss with Cotton